incubator/hnc/internal/validators/object.go

```diff
@@ -275,16 +276,19 @@ func (o *Object) handleInherited(op k8sadm.Operation, newSource, oldSource strin
 	return deny(metav1.StatusReasonInternalError, "unknown operation: "+string(op))
 }
 
-// isDeletingNS returns true if the namespace of the object is already being deleted, based on the
-// in-memory forest.
-func (o *Object) isDeletingNS(inst *unstructured.Unstructured) bool {
-	o.Forest.Lock()
-	defer o.Forest.Unlock()
-	ns := o.Forest.Get(inst.GetNamespace())
-	if ns == nil {
-		return false
+// validateDeletingNS validates if the namespace of the object is already being deleted
+func (o *Object) validateDeletingNS(ctx context.Context, inst *unstructured.Unstructured) error {
+	ns := &corev1.Namespace{}
+	nnm := types.NamespacedName{Name: inst.GetNamespace()}
+	if err := o.client.Get(ctx, nnm, ns); err != nil {
+		return err
```

Comment:
Wrap this error: `return fmt.Errorf("while determining whether namespace %q is being deleted: %w", inst.GetNamespace(), err)`

Reply:
Also: I've seen some weird corner cases where the namespace can be deleted even before the objects are. So maybe let's treat not-found errors specially? For an example, see: https://github.com/kubernetes-sigs/multi-tenancy/blob/d7e8a4dbcf5e8e594956a0546c73b015841e1323/incubator/hnc/internal/validators/hierarchy.go#L465